cli/cmd/generate.go

```diff
@@ -0,0 +1,461 @@
+			}
+			if err := validateOutputLocation(dirname); err != nil {
+				return err
+			}
+
+			// Validate aws profile, if passed
+			profile, err := cmd.Flags().GetString("aws_profile")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateAwsProfile(profile); profile != "" && err != nil {
+				return err
+			}
+
+			// Validate aws region, if passed
+			region, err := cmd.Flags().GetString("aws_region")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateAwsRegion(region); region != "" && err != nil {
+				return err
+			}
+
+			// Validate cloudtrail bucket arn, if passed
+			arn, err := cmd.Flags().GetString("existing_bucket_arn")
+			if err != nil {
+				return errors.Wrap(err, "failed to load command flags")
+			}
+			if err := validateAwsArnFormat(arn); arn != "" && err != nil {
+				return err
+			}
+
+			// Load any cached inputs if interactive
+			if cli.InteractiveMode() {
+				cachedOptions := &aws.GenerateAwsTfConfigurationArgs{}
+				if ok := cli.ReadCachedAsset(CachedAssetIacParams, &cachedOptions); ok {
+					cli.Log.Debug("loaded previously set values for AWS iac generation")
+				}
+
+				extraState := &AwsGenerateCommandExtraState{}
+				if ok := cli.ReadCachedAsset(CachedAssetAwsExtraState, &extraState); ok {
+					cli.Log.Debug("loaded previously set values for AWS iac generation (extra state)")
+				}
+
+				// Merge cached inputs to current options (current options win)
+				if err := mergo.Merge(GenerateAwsCommandState, cachedOptions); err != nil {
+					return errors.Wrap(err, "failed to load saved options")
+				}
+				if err := mergo.Merge(GenerateAwsCommandExtraState, extraState); err != nil {
+					return errors.Wrap(err, "failed to load saved options")
+				}
+			}
+
+			// Parse passed in subaccounts (if any)
+			if len(GenerateAwsCommandExtraState.AwsSubAccounts) > 0 {
+				// validate consolidated_cloudtrail is enabled - otherwise this flag cannot be used
+				if ok, _ := cmd.Flags().GetBool("consolidated_cloudtrail"); !ok {
+					return errors.New("aws subaccounts can only be supplied with consolidated cloudtrail enabled")
+				}
+
+				// validate the format of supplied values is correct
+				if err := validateAwsSubAccounts(GenerateAwsCommandExtraState.AwsSubAccounts); err != nil {
+					return err
+				}
+
+				awsSubAccounts := []aws.AwsSubAccount{}
+				for _, account := range GenerateAwsCommandExtraState.AwsSubAccounts {
+					accountDetails := strings.Split(account, ":")
+					awsSubAccounts = append(awsSubAccounts, aws.NewAwsSubAccount(accountDetails[0], accountDetails[1]))
+				}
+				GenerateAwsCommandState.SubAccounts = awsSubAccounts
+			}
+
+			// Collect and/or confirm parameters
+			err = promptAwsGenerate(GenerateAwsCommandState, GenerateAwsExistingRoleState, GenerateAwsCommandExtraState)
+			if err != nil {
+				return errors.Wrap(err, "error when collecting/confirming parameters")
```

Comment:
This looks odd since it says error twice
```
lacework cloud-account iac-generate aws
▸ Enable Config Integration? No
▸ Enable Cloudtrail Integration? No

ERROR error when collecting/confirming parameters: must enable cloudtrail or config
```
```suggestion
				return errors.Wrap(err, "collecting/confirming parameters")
```

Reply:
👍